### PR TITLE
remove metastore calls in leaves

### PR DESCRIPTION
**Pull request by PSeitz**
- pass index config to leaf nodes
- pass index_uri to leaf, remove metastore call
- remove metastore call in fetch_docs
- remove metastore call in stream request

### Context / purpose
Describe the context and purpose of this PR. Link related issue(s) if any.

### Description
Describe the proposed changes made in this PR.

### How was this PR tested?
Describe how you tested this PR.

### Checklist
*Remove or complete this list if required.*
- [ ] tested locally
- [ ] added unit tests
- [ ] included documentation

### CODE REVIEW

File: quickwit-index-config/src/config.rs [hunk 1]
```diff
@@ -57,6 +57,14 @@ pub enum SortBy {
     DocId,
 }
 
+/// An index metadata carries all meta data about an index.
```
Comment (fmassot) [inline quickwit-index-config/src/config.rs:60]: no struct finally ;)

Comment (fulmicoton) [reply quickwit-index-config/src/config.rs:60]:
```suggestion
```

File: quickwit-index-config/src/config.rs [hunk 2]
```diff
@@ -57,6 +57,14 @@ pub enum SortBy {
...
+//#[derive(Clone, Debug, Serialize, Deserialize)]
```
Comment (fulmicoton) [inline quickwit-index-config/src/config.rs:61]:
```suggestion
//#[derive(Clone, Debug, Serialize, Deserialize)]
```

Comment (fulmicoton) [inline quickwit-index-config/src/config.rs:61]:
```suggestion
//#[derive(Clone, Debug, Serialize, Deserialize)]
```

File: quickwit-index-config/src/config.rs [hunk 3]
```diff
@@ -57,6 +57,14 @@ pub enum SortBy {
...
+//pub struct IndexConfigo {
+//schema: Schema,
+//default_sort_by: SortBy,
+//timestamp_field: Option<Field>,
+//}
+
```
Comment (fulmicoton) [inline quickwit-index-config/src/config.rs:67]:
```suggestion
```

**Review by fulmicoton — APPROVED** [review]

File: quickwit-index-config/src/config.rs [hunk 4]
```diff
@@ -152,4 +152,30 @@ mod tests {
...
+        let deserialized_default_config =
+            serde_json::from_str::<Box<dyn IndexConfig>>(JSON_DEFAULT_INDEX_CONFIG)?;
+        let expected_default_config = DefaultIndexConfigBuilder::new().build()?;
+        assert_eq!(
+            format!("{:?}", deserialized_default_config),
+            format!("{:?}", expected_default_config),
+        );
+
+        let hmm_ok = serde_json::to_string(&deserialized_default_config)?;
```
Comment (fmassot) [inline quickwit-index-config/src/config.rs:173]: better name? like serialized_config?

Comment (PSeitz) [reply quickwit-index-config/src/config.rs:173]: hmm ok .. :)

**Review by fmassot — APPROVED** [review]

File: quickwit-index-config/src/config.rs (hunk 2 above)
Comment (guilload) [inline quickwit-index-config/src/config.rs:61]: I wish we removed that commented-out code before merging the PR t_t

Comment (PSeitz) [reply quickwit-index-config/src/config.rs:61]: I think Paul removed it, but I accidentally restored it with a rebase + force push

**Pull request merged**